src/component/linechart/index.tsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react'
+import { AreaChart, Area, Tooltip, XAxis } from 'recharts' 
+
+type DataItem = {
+  [key:string]: string | number
+}
+
+interface ChartProps {
+  dataKey: string,
+  data: DataItem[],
+}
+
+const LineChart: React.FC<ChartProps> = ({ data, dataKey }) => {
```

Comment:
We talked about component structure so that we separate behaviour and style. One idea below.  

```
$ tree
.
└── src
    └── component
        └── linechart
            ├── behaviour
            └── design
```